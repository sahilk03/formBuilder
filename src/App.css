.maincontainer{
  height: 100vh;
  background: burlywood;
  /* display: flex;
  flex-direction: column; */
}
.innercontainer{
  display: flex;
  flex-direction: row;
  height: 100%;
  align-items: center;
  justify-content: center;
}
.card{
  flex-grow: 1;
  background: white;
  height: 80%;
  border-radius: 5px;
  max-width: 40%;
  margin: 10px;
  padding: 10px;
  overflow-y: auto;
}
.mainLabel{
  display: block;
  font-size: 1.17em;
  margin-top: 1em;
  margin-bottom: 0.2em;
  margin-left: 0;
  margin-right: 0;
  font-weight: bold;
}
.nocontent{
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 1.17em;
    color: rgb(202, 196, 196);
}