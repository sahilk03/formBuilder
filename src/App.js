import React, { useState } from "react";
import "./App.css";
import Checkboxwithlabel from "./Checkboxwithlabel";
import SelectedComponent from "./SelectedComponent";
function App() {
  const [selectedList, setSelectedList] = useState({});
  function onChange(event, label, mainLabel) {
    if (event.target.checked) {
      setSelectedList((prevState) => {
        if (prevState[mainLabel]) {
          return {
            ...prevState,
            [mainLabel]: [...prevState[mainLabel], label],
          };
        } else {
          return { ...prevState, [mainLabel]: [label] };
        }
      });
    } else {
      setSelectedList((prevState) => {
        debugger;
        if (prevState[mainLabel]) {
          let tempArr = [...prevState[mainLabel]].filter((l) => l != label);
          if (tempArr.length < 1) {
            let result = { ...prevState };
            delete result[mainLabel];
            return result;
          }
          return {
            ...prevState,
            [mainLabel]: tempArr,
          };
        }
      });
    }
  }
  return (
    <div className="maincontainer">
      <div className="innercontainer">
        <div className="card">
          <div className="form-check" key="Portugal">
            <div className="mainLabel">Portugal</div>
            {[
              "Aasiya Jayavant",
              "Luvleen Lawrence",
              "Rey Mibourne",
              "Cayla Brister",
            ].map((label) => (
              <Checkboxwithlabel
                mainLabel={"Portugal"}
                label={label}
                onChange={onChange}
                key={label}
              />
            ))}
          </div>
          <div className="form-check" key="Nicaragua">
            <div className="mainLabel">Nicaragua</div>
            {[
              "Deveedaas Nandi",
              "Obasey Chidy",
              "Xenie Dolezelova",
              "Ezequiel Dengra",
            ].map((label) => (
              <Checkboxwithlabel
                mainLabel={"Nicaragua"}
                label={label}
                onChange={onChange}
                key={label}
              />
            ))}
          </div>
          <div className="form-check" key="Marshall">
            <div className="mainLabel">Marshall Islands</div>
            {["Aaron Almaraz", "Jelena Denisova"].map((label) => (
              <Checkboxwithlabel
                mainLabel={"Marshall Islands"}
                label={label}
                onChange={onChange}
                key={label}
              />
            ))}
          </div>
        </div>
        <div className="card">
          <SelectedComponent selectedList={selectedList} />
        </div>
      </div>
    </div>
  );
}

export default App;
