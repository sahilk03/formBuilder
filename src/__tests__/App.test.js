import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

describe("testing SelectedComponent", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
