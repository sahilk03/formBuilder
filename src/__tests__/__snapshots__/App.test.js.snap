// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SelectedComponent matches snapshot 1`] = `
<div
  className="maincontainer"
>
  <div
    className="innercontainer"
  >
    <div
      className="card"
    >
      <div
        className="form-check"
      >
        <div
          className="mainLabel"
        >
          Portugal
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Aasiya Jayavant"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Aasiya Jayavant"
            htmlFor="Aasiya Jayavant"
          >
            Aasiya Jayavant
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Luvleen Lawrence"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Luvleen Lawrence"
            htmlFor="Luvleen Lawrence"
          >
            Luvleen Lawrence
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Rey Mibourne"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Rey Mibourne"
            htmlFor="Rey Mibourne"
          >
            Rey Mibourne
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Cayla Brister"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Cayla Brister"
            htmlFor="Cayla Brister"
          >
            Cayla Brister
          </label>
        </div>
      </div>
      <div
        className="form-check"
      >
        <div
          className="mainLabel"
        >
          Nicaragua
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Deveedaas Nandi"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Deveedaas Nandi"
            htmlFor="Deveedaas Nandi"
          >
            Deveedaas Nandi
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Obasey Chidy"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Obasey Chidy"
            htmlFor="Obasey Chidy"
          >
            Obasey Chidy
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Xenie Dolezelova"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Xenie Dolezelova"
            htmlFor="Xenie Dolezelova"
          >
            Xenie Dolezelova
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Ezequiel Dengra"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Ezequiel Dengra"
            htmlFor="Ezequiel Dengra"
          >
            Ezequiel Dengra
          </label>
        </div>
      </div>
      <div
        className="form-check"
      >
        <div
          className="mainLabel"
        >
          Marshall Islands
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Aaron Almaraz"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Aaron Almaraz"
            htmlFor="Aaron Almaraz"
          >
            Aaron Almaraz
          </label>
        </div>
        <div
          className="checkboxblock"
        >
          <input
            className="form-check-input checkbox"
            id="Jelena Denisova"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            data-testid="Jelena Denisova"
            htmlFor="Jelena Denisova"
          >
            Jelena Denisova
          </label>
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="nocontent"
      >
        No Value Selected
      </div>
    </div>
  </div>
</div>
`;
